shell/gpu/gpu_surface_gl_skia.h

```diff
@@ -56,23 +57,37 @@ class GPUSurfaceGLSkia : public Surface {
   bool AllowsDrawingWhenGpuDisabled() const override;
 
  private:
-  bool CreateOrUpdateSurfaces(const SkISize& size);
-
-  sk_sp<SkSurface> AcquireRenderSurface(
-      const SkISize& untransformed_size,
-      const SkMatrix& root_surface_transformation);
-
-  bool PresentSurface(const SurfaceFrame& frame, DlCanvas* canvas);
+  struct ViewUnit {
+    ViewUnit(sk_sp<SkSurface> onscreen_surface,
+             uint32_t fbo_id,
+             std::optional<SkIRect> existing_damage)
+        : onscreen_surface(std::move(onscreen_surface)),
+          fbo_id(fbo_id),
+          existing_damage(existing_damage) {}
+
+    sk_sp<SkSurface> onscreen_surface;
+    /// FBO backing the current `onscreen_surface`.
+    uint32_t fbo_id = 0;
+    // The FBO's existing damage, as tracked by the GPU surface, delegates still
+    // have an option of overriding this damage with their own in
+    // `GLContextFrameBufferInfo`.
+    std::optional<SkIRect> existing_damage = std::nullopt;
+  };
+
+  bool CreateOrUpdateSurfaces(std::unique_ptr<ViewUnit>& view_unit,
+                              const SkISize& size);
+
+  ViewUnit* AcquireRenderSurface(uint64_t view_id,
+                                 const SkISize& untransformed_size,
+                                 const SkMatrix& root_surface_transformation);
+
+  bool PresentSurface(uint64_t view_id,
+                      const SurfaceFrame& frame,
+                      DlCanvas* canvas);
 
   GPUSurfaceGLDelegate* delegate_;
   sk_sp<GrDirectContext> context_;
-  sk_sp<SkSurface> onscreen_surface_;
-  /// FBO backing the current `onscreen_surface_`.
-  uint32_t fbo_id_ = 0;
-  // The current FBO's existing damage, as tracked by the GPU surface, delegates
-  // still have an option of overriding this damage with their own in
-  // `GLContextFrameBufferInfo`.
-  std::optional<SkIRect> existing_damage_ = std::nullopt;
+  std::map<uint64_t, std::unique_ptr<ViewUnit>> view_units_;
```

Comment:
The ownership around the values in this map is a little bit murky. Can you try to clean that up a bit?

Consider using shared pointers if you need multiple methods to be able to call non-const methods on this pointer. 

Right now it seems pretty error prone to me that a particular method could leave the map with a null view_unit.